lib/rouge/lexers/xquery.rb

```diff
@@ -0,0 +1,143 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+module Rouge
+  module Lexers
+    load_lexer 'xpath.rb'
+    class XQuery < XPath
+      title 'XQuery'
+      desc 'XQuery 3.1: An XML Query Language'
+      tag 'xquery'
+      filenames '*.xquery', '*.xq'
+      mimetypes 'application/xquery'
+
+      def self.keywords
+        @keywords ||= Regexp.union super, Regexp.union(%w(
+          xquery encoding version declare module
+          namespace copy-namespaces boundary-space construction
+          default collation base-uri preserve strip
+          ordering ordered unordered order empty greatest least
+          preserve no-preserve inherit no-inherit
+          decimal-format decimal-separator grouping-separator
+          infinity minus-sign NaN percent per-mille
+          zero-digit digit pattern-separator exponent-separator
+          import schema at element option
+          function external context item
+          typeswitch switch case
+          try catch
+          validate lax strict type
+          document element attribute text comment processing-instruction
+          for let where order group by return
+          allowing tumbling stable sliding window
+          start end only when previous next count collation
+          ascending descending
+        ))
+      end
+
+      state :tags do
+        rule %r/<#{XPath.qName}/, Name::Tag, :start_tag
+        rule %r/<!--/, Comment, :xml_comment
+        rule %r/<\?.*?\?>/, Comment::Preproc
+        rule %r/<!\[CDATA\[.*?\]\]>/, Comment::Preproc
+        rule %r/&\S*?;/, Name::Entity
+      end
+
+      prepend :root do
+        mixin :tags
+
+        rule %r/{/, Punctuation, :root
```

Comment:
Are you pushing `root` here because you need to pop closing braces because of how you're using `:root` with `:tag_content`?

Reply:
Separately, the brace should be escaped.

Reply:
Yes, that's it. In a number of locations, I push `:root` because `{...}` is code interpolation. But you're right: in this case, it should actually be safe to just stay in `:root` rather than push an identical state, because the pop for closing brackets `}` checks that the stack won't be empty anyway. I've fixed this in 511e19a60fd97c790407a9acf5cf9b1552bc31b4

Reply:
The braces have been escaped in 70631b9a85a82af1180bcf4cdca5a8ebd86cf082.